### ISSUE TITLE
failed installation on NVIDIA JETSON TX2 (ARM64)

### ISSUE THREAD

**Issue by Icy0217** [taichi-dev/taichi#2287]
**Describe the bug**
Build from source on NVIDIA JETSON TX2 (ARM64)
Ubuntu 18.04
CUDA 10.0
Python 3.6.9 
cmake 3.10.2

**Error Log**
```py
nvidia@nvidia-desktop:~/taichi/build$ cmake .. -DCMAKE_CXX_COMPILER=clang-8
-- The CXX compiler identification is Clang 8.0.0
-- Check for working CXX compiler: /usr/bin/clang-8
-- Check for working CXX compiler: /usr/bin/clang-8 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Taichi Version 0.7.16
       Commit 940fa075b3eacf0be18dee3cdeadc073eab9bdd7
Using 'python3' as python interpreter.
Python 3.6.9
    version: 3.6
    include: /usr/include/python3.6m
    library: PYTHON_LIBRARY-NOTFOUND
    numpy include: /usr/local/lib/python3.6/dist-packages/numpy/core/include
    pybind11 include: /usr/local/lib/python3.6/dist-packages/pybind11/include;/usr/local/lib/python3.6/dist-packages/pybind11/include
Using C++ compiler: /usr/bin/clang-8
Clang compiler detected. Using std=c++17.
Building for processor aarch64
Using float32 (single) precision as real
Building with GLFW
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Using X11 for window creation
-- Looking for XOpenDisplay in /usr/lib/aarch64-linux-gnu/libX11.so;/usr/lib/aarch64-linux-gnu/libXext.so
-- Looking for XOpenDisplay in /usr/lib/aarch64-linux-gnu/libX11.so;/usr/lib/aarch64-linux-gnu/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found X11: /usr/lib/aarch64-linux-gnu/libX11.so
CMake Error at cmake/TaichiCore.cmake:128 (target_link_libraries):
  Object library target "taichi_isolated_core" may not link to anything.
Call Stack (most recent call first):
  CMakeLists.txt:53 (include)


CMake Warning (dev) in CMakeLists.txt:
  No cmake_minimum_required command is present.  A line of code such as

    cmake_minimum_required(VERSION 3.10)

  should be added at the top of the file.  The version specified may be lower
  if you wish to support older CMake versions for this project.  For more
  information run "cmake --help-policy CMP0000".
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring incomplete, errors occurred!
See also "/home/nvidia/taichi/build/CMakeFiles/CMakeOutput.log".
See also "/home/nvidia/taichi/build/CMakeFiles/CMakeError.log".

```

**Comment by k-ye** [comment]
Please upgrade your `cmake` to 3.12+

**Comment by Icy0217** [comment]
Hi @k-ye , Thanks for help! Problem solved!
But a new problem occur as below when I compile the code:
```py
nvidia@nvidia-desktop:~/taichi/build$ make -j8
Scanning dependencies of target generate_commit_hash
Scanning dependencies of target glfw
[  0%] Building C object external/glfw/src/CMakeFiles/glfw.dir/input.c.o
[  0%] Building C object external/glfw/src/CMakeFiles/glfw.dir/context.c.o
[  1%] Building C object external/glfw/src/CMakeFiles/glfw.dir/init.c.o
[  2%] Building C object external/glfw/src/CMakeFiles/glfw.dir/window.c.o
......
[ 95%] Built target taichi_isolated_core
Scanning dependencies of target taichi_core
[ 95%] Building CXX object CMakeFiles/taichi_core.dir/taichi/python/exception.o
[ 96%] Building CXX object CMakeFiles/taichi_core.dir/taichi/python/export.o
[ 97%] Building CXX object CMakeFiles/taichi_core.dir/taichi/python/export_lang.o
[ 97%] Building CXX object CMakeFiles/taichi_core.dir/taichi/python/export_visual.o
[ 98%] Building CXX object CMakeFiles/taichi_core.dir/taichi/python/export_misc.o
[ 99%] Building CXX object CMakeFiles/taichi_core.dir/taichi/python/interfaces_registry.o
[ 99%] Building CXX object CMakeFiles/taichi_core.dir/taichi/python/export_math.o
[ 99%] Building CXX object CMakeFiles/taichi_core.dir/taichi/python/memory_usage_monitor.o
[100%] Building CXX object CMakeFiles/taichi_core.dir/taichi/python/py_exception_translator.o
[100%] Linking CXX shared library libtaichi_core.so
/usr/bin/ld: libtaichi_core.so: No symbol version section for versioned symbol `powf@GLIBC_2.2.5'
/usr/bin/ld: final link failed: Nonrepresentable section on output
clang: error: linker command failed with exit code 1 (use -v to see invocation)
CMakeFiles/taichi_core.dir/build.make:555: recipe for target 'libtaichi_core.so' failed
make[2]: *** [libtaichi_core.so] Error 1
CMakeFiles/Makefile2:104: recipe for target 'CMakeFiles/taichi_core.dir/all' failed
make[1]: *** [CMakeFiles/taichi_core.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2


```

**Comment by k-ye** [comment]
Hmm, could you try `cmake %whatever_your_original_args_are -DTI_WITH_OPENGL=OFF`?

**Comment by Icy0217** [comment]
Thanks @k-ye ! I use the issue #377 and `cmake ..  -DCMAKE_C_COMPILER=/usr/bin/clang-8 -DCMAKE_CXX_COMPILER=/usr/bin/clang++-8 -DTI_WITH_OPENGL=OFF`. It built successfully, but encounter error when I run the example.

```py
nvidia@nvidia-desktop:~/taichi/examples$ python3 mpm128.py
[Taichi] mode=development
[Taichi] preparing sandbox at /tmp/taichi-8g3_hl5t
Share object taichi_core import failed, check this page for possible solutions:
https://taichi.readthedocs.io/en/stable/install.html#troubleshooting
Taichi core import failed: /tmp/taichi-8g3_hl5t/taichi_core.so: undefined symbol: __wrap_log2f
Consider installing libtinfo5: sudo apt-get install libtinfo5
check this page for possible solutions:
https://taichi.readthedocs.io/en/stable/install.html#troubleshooting
Traceback (most recent call last):
  File "mpm128.py", line 1, in <module>
    import taichi as ti
  File "/home/nvidia/taichi/python/taichi/__init__.py", line 1, in <module>
    from .core import *
  File "/home/nvidia/taichi/python/taichi/core/__init__.py", line 1, in <module>
    from taichi.core.logging import *
  File "/home/nvidia/taichi/python/taichi/core/logging.py", line 4, in <module>
    from taichi.core import util
  File "/home/nvidia/taichi/python/taichi/core/util.py", line 219, in <module>
    raise e from None
  File "/home/nvidia/taichi/python/taichi/core/util.py", line 211, in <module>
    import_ti_core(tmp_dir)
  File "/home/nvidia/taichi/python/taichi/core/util.py", line 48, in import_ti_core
    raise e from None
  File "/home/nvidia/taichi/python/taichi/core/util.py", line 36, in import_ti_core
    import taichi_core as core
ImportError: /tmp/taichi-8g3_hl5t/taichi_core.so: undefined symbol: __wrap_log2f
Consider installing libtinfo5: sudo apt-get install libtinfo5


```

It seems libtinfo5 is required, but it is already installed.
```py

nvidia@nvidia-desktop:~/taichi/examples$ sudo apt-get install libtinfo5
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libtinfo5 is already the newest version (6.1-1ubuntu1.18.04).
0 upgraded, 0 newly installed, 0 to remove and 143 not upgraded.


```

**Comment by k-ye** [comment]
What happens if you comment out this part?

https://github.com/taichi-dev/taichi/blob/7e54f56d305f370f773264864f7fcf117470ddf1/taichi/common/core.cpp#L13-L23

**Comment by bobcao3** [comment]
This should be fixed now, redirecting to #4711
